lib/avromatic/model/logical_types.rb

```diff
@@ -0,0 +1,16 @@
+module Avromatic
+  module Model
+    module LogicalTypes
+
+      LOGICAL_TYPE_MAP = {
+        'date' => Date,
+        'timestamp-micros' => Time,
+        'timestamp-millis' => Time
+      }.freeze
+
+      def self.value_class(logical_type)
+        LOGICAL_TYPE_MAP.fetch(logical_type, nil)
```

Comment:
This could probably just be `LOGICAL_TYPE_MAP[logical_type]` since you're falling back to `nil` anyway.